src/java/org/apache/cassandra/replication/MutationTrackingService.java

```diff
@@ -915,6 +978,62 @@ static List<KeyspaceShards> loadFromSystemTables(ClusterMetadata cluster, LongSu
         }
     }
 
+    private static class CoordinatedTransfers implements Iterable<CoordinatedTransfer>
```

Comment:
I don't think this needs to be a private inner class. It could also use some unit tests checking the sstable selection logic